docker/autoscaler/Dockerfile

```diff
@@ -0,0 +1,13 @@
+ARG BASE_IMAGE
+FROM rayproject/ray:nightly"$BASE_IMAGE"
+
+COPY .whl/ray-2.0.0.dev0-cp37-cp37m-manylinux2014_x86_64.whl .
```

Comment:
Is this for testing purpose? Trying to understand if `run_autoscaler.py` and `run_autoscaler_with_retries.py` will be distributed with ray core together once PR is merged?

Reply:
Yes this `COPY` is for testing, will be removed before merging. `run_autoscaler.py` will be distributed with Ray and I'll try to do the retries on the pod level before merging this.

Reply:
I need to read the code more carefully but just a comment on pod-level retries that we want to avoid crash-looping if possible -- we'd get unnecessarily slow start-up.

Reply:
(Due to exponential backoff.)

Reply:
Yes, this is a very good point. I think given that we have been using `run_autoscaler_with_retries.py` for a while in Anyscale to run the Ray Autoscaler and it has been working very well, let's keep it for this PR.

I did fix the other development changes that were still in the Dockerfile.